application/src/main/java/org/opentripplanner/apis/gtfs/datafetchers/stepImpl.java

```diff
@@ -65,16 +64,14 @@ public DataFetcher<Object> feature() {
       WalkStep walkStep = getSource(environment);
       if (walkStep.entrance().isPresent()) {
         return walkStep.entrance().get();
-      } else if (walkStep.getEdges().size() == 1) {
-        VerticalTransportationUse verticalTransportationUse = environment
-          .<GraphQLRequestContext>getContext()
-          .verticalTransportationUseFactory()
-          .createInclinedVerticalTransportationUse(walkStep.getEdges().getFirst());
-        if (verticalTransportationUse != null) {
-          return switch (verticalTransportationUse) {
-            case EscalatorUse escalatorUse -> escalatorUse;
-            case StairsUse stairsUse -> stairsUse;
-          };
+      } else if (walkStep.verticalTransportationUse().isPresent()) {
+        VerticalTransportationUse verticalTransportationUse = walkStep
+          .verticalTransportationUse()
+          .get();
+        if (verticalTransportationUse instanceof EscalatorUse escalatorUse) {
+          return escalatorUse;
+        } else if (verticalTransportationUse instanceof StairsUse stairsUse) {
+          return stairsUse;
```

Comment:
You could still use the switch here. On the other hand, are these ifs even necessary, could you just return verticalTransportationUse? The StepFeatureTypeResolver should handle the supported types.

Reply:
I removed this, good catch